src/sentry/auth/idpmigration.py

```diff
@@ -15,8 +38,18 @@ def create_verification_key(user: User, org: Organization, email: str) -> None:
     :param org: the organization whose SSO provider is being used
     :param email: the email address associated with the SSO identity
     """
-    key = str(uuid.uuid4())  # noqa
-    raise NotImplementedError  # TODO
+    redis_key = "verificationKeyStorage"
+    TTL = 60 * 10
+    cluster = redis.clusters.get("default").get_local_client_for_key(redis_key)
+    member_id = OrganizationMember.objects.get(organization=org, user=user).id
+
+    verification_key = f"auth:one-time-key:{uuid4().hex}"
```

Comment:
looking at this again, it may be better to not use a uuidv4. https://security.stackexchange.com/questions/157270/using-v4-uuid-for-authentication

instead, you can use the method `default_validation_hash` from useremail.py

Reply:
Does the email need the verification_key with the `auth:one-time-key` prefix?

Reply:
This is an interesting mini-rabbit hole. I believe a UUID v4 actually _would_ be secure, but only because

1. it's v4 and not a different version that uses derived state that an attacker could feasibly guess, and
2. Python's `uuid.uuid4()` implementation uses a cryptographically secure source of random bytes, unlike some of the crappier UUID4 utilities cited in the Stack Exchange answer.

That said, I'm in favor of not using a UUID because it's misleading -- we're not using it merely to identify something, but as a verification key. 👍 to doing something like `default_validation_hash`.